### PR TITLE
Implement new-rollup staker logic

### PR DESCRIPTION
**Pull request by PlasmaPower**

### CODE REVIEW

File: packages/arb-node-core/staker/validator.go [hunk 1]
```diff
@@ -108,133 +108,146 @@ func (v *Validator) resolveNextNode(ctx context.Context) error {
...
 }
 
-type nodeActionInfo interface {
-}
+type nodeAction interface{}
 
-func (v *Validator) generateNodeAction(ctx context.Context, base core.NodeID, maybeMakeNode bool) (nodeActionInfo, error) {
-	startState, err := lookupNodeStartState(ctx, v.rollup.RollupWatcher, base)
+func (v *Validator) generateNodeAction(ctx context.Context, address common.Address, active bool, proactiveNewNodes bool) (nodeAction, bool, error) {
+	base, baseHash, err := v.validatorUtils.LatestStaked(ctx, address)
```
Comment (hkalodner) [inline packages/arb-node-core/staker/validator.go:126]: I think that as implemented, the watchtower strategy will always act as though it's on the latest confirmed. I think we need to instead find invalid nodes when they're initially posted

**Review by hkalodner — CHANGES_REQUESTED** [review]

File: packages/arb-node-core/ethbridge/rollupWatcher.go [hunk 2]
```diff
@@ -103,39 +103,81 @@ func (r *RollupWatcher) LookupCreation(ctx context.Context) (*ethbridgecontracts
...
 	}
 	infos := make([]*core.NodeInfo, 0, len(logs))
-	for _, ethLog := range logs {
+	lastHash := parentHash
+	for i, ethLog := range logs {
 		parsedLog, err := r.con.ParseNodeCreated(ethLog)
 		if err != nil {
 			return nil, err
 		}
+		if parsedLog.ParentNodeHash != parentHash {
```
[the remark this answers is not in the document]
Comment (PlasmaPower) [reply packages/arb-node-core/ethbridge/rollupWatcher.go]: This was before I knew how log querying worked - I think it's fine, we need to trust our eth node anyways.

File: packages/arb-node-core/staker/staker_test.go [hunk 3]
```diff
@@ -111,40 +116,91 @@ func TestStaker(t *testing.T) {
...
 
-	_, err = val.ExecuteTransactions(ctx, staker.builder)
-	test.FailIfError(t, err)
+	faultyStakerAlive := false
+	faultyStakerDead := false
+	for i := 100; i >= 0; i++ {
+		if (i % 2) == 0 {
+			_, err := staker.Act(ctx)
+			test.FailIfError(t, err)
+		} else if !faultyStakerAlive || !faultyStakerDead {
```
Comment (hkalodner) [inline packages/arb-node-core/staker/staker_test.go:168]: Schrödinger's staker?

**Review by hkalodner — APPROVED** [review]
LGTM

**Pull request merged**